packages/ui/src/proposals/modals/AddNewProposal/components/SpecificParameters/RuntimeUpgrade.tsx

```diff
@@ -24,6 +25,22 @@ const MAX_FILE_SIZE = 3 * 1024 * 1024
 const isDragEvent = (event: any): event is React.DragEvent<HTMLElement> => !!event?.dataTransfer
 const isChangeEvent = (event: any): event is React.ChangeEvent<HTMLInputElement> => !!event?.target?.files
 
+// https://github.com/paritytech/substrate/blob/master/primitives/maybe-compressed-blob/src/lib.rs
+// Look for 8 byte magic prefix in blob that indicates it is compressed with zstd algorithm
+// If is compressed, strips the prefix and decompresses remaining bytes, otherwise returns original blob
+const maybeDecompressRuntimeBlob = async (blob: ArrayBuffer): Promise<Buffer> => {
+  const ZSTD_PREFIX = Buffer.from([82, 188, 83, 118, 70, 219, 142, 5])
+  let wasm = Buffer.from(blob)
+  const prefix = wasm.slice(0, 8)
+  const isCompressed = Buffer.compare(prefix, ZSTD_PREFIX) === 0
+  if (isCompressed) {
+    const { ZstdStream } = await ZstdInit()
```

Comment:
I suspect this is not he best place to import or load the library, but I'm not an expert in React so perhaps it can be refactored by someone else, or suggest how/where to place initializing of the library and how to pass it down to this component.

Was also looking for a way to "unload" the library from memory (as I have seem some other wasm libraries do) but couldn't find it in the documentation.

Reply:
I almost forgot about this! You're right it makes sense to move this lib out of the main bundle.

I chose to move it to a web worker in [this commit](https://github.com/Joystream/pioneer/pull/4182/commits/d35c3c6b5f7b229560c0478bad67b843811408f9) (because it's already a common pattern in Pioneer) however I'm not sure where did the library actually went:
![Screenshot from 2023-05-12 14-37-51](https://github.com/Joystream/pioneer/assets/6571453/867dd565-e8f5-482c-986d-727425e15e43)

The main bundle lost the correct size, but I couldn't find any bundles which significantly grew. Maybe the lib was already a dependency in that chunk.